binja_script.py

```diff
@@ -0,0 +1,400 @@
+        ty = ty.target
+        derefcnt += 1
+    return (str(ty), derefcnt)
+
+
+to_register = []
+
+
+def should_register(f):
+    to_register.append(f.__name__)
+    return f
+
+
+class ServerHandler:
+    bv: binaryninja.BinaryView
+
+    def __init__(self, bv: binaryninja.BinaryView):
+        self.bv = bv
+
+    # initialize a binaryview if not already initialized, e.g. add a tag type
+    def init(self) -> None:
+        tag_types = {"pwndbg-pc": "➡️", "pwndbg-bp": "🔴"}
+        for k, v in tag_types.items():
+            if k not in self.bv.tag_types:
+                self.bv.create_tag_type(k, v)
+
+    @should_register
+    def clear_pc_tag(self) -> None:
+        """
+        Clear all instances of the 'current pc' tag.
+        """
+        for t in get_tag_refs(self.bv, "pwndbg-pc"):
+            remove_tag_ref(self.bv, t)
+
+    @should_register
+    def navigate_to(self, addr: int) -> None:
+        """
+        Navigate to a specified address.
+        """
+        self.bv.navigate(self.bv.view, addr)
+
+    @should_register
+    def update_pc_tag(self, new_pc: int) -> None:
+        """
+        Sets the 'current pc' tag to the specified address, and clears the old ones.
+        """
+        self.clear_pc_tag()
+        add_auto_tag(self.bv, new_pc, "pwndbg-pc", "current pc")
+
+    @should_register
+    def get_bp_tags(self) -> List[int]:
+        """
+        Gets a list of all addresses with a breakpoint tag.
+        """
+        return [t.addr for t in get_tag_refs(self.bv, "pwndbg-bp")]
+
+    @should_register
+    def get_symbol(self, addr: int) -> str | None:
+        """
+        Gets the symbol at exactly the specified address.
+        """
+        sym = self.bv.get_symbol_at(addr)
+        if sym is None:
+            return None
+        return sym.full_name
+
+    @should_register
+    def get_func_info(self, addr: int) -> Tuple[str, int] | None:
+        """
+        Gets the widest function containing the specified address.
+
+        Returns a (function name, offset from start) tuple.
+        """
+        func = get_widest_func(self.bv, addr)
+        if func is None:
+            return None
+        return (func.symbol.full_name, func.start)
+
+    @should_register
+    def get_data_info(self, addr: int) -> Tuple[str, int] | None:
+        """
+        Gets the data variable containing the specified address.
+
+        Returns a (variable name, offset from start) tuple.
+        """
+        dv = self.bv.get_data_var_at(addr)
+        if dv is None:
+            return None
+        if dv.symbol is not None:
+            return (dv.symbol.full_name, dv.address)
+        return (dv.name or f"data_{dv.address:x}", dv.address)
+
+    @should_register
+    def get_comments(self, addr: int) -> List[str]:
+        """
+        Gets a list of all comments at a specified address.
+        """
+        ret = []
+        for f in sorted(self.bv.get_functions_containing(addr), key=lambda f: f.start):
```

Comment:
Note: `key=lambda f: f.start` the `f` is confusing here since we do `for f in sorted(...)`, lets maybe rename one of them to `func`